data_model/src/permission/mod.rs

```diff
@@ -26,13 +21,13 @@ pub use token::Token;
 pub use validator::Validator;
 
 /// Collection of [`Token`]s
-pub type Permissions = btree_set::BTreeSet<token::Token>;
+pub type Permissions = Vec<token::Token>;
```

Comment:
?

Reply:
Oh, yes, this change is not obvious.

So when I introduced `Validator` into `Value` I made `Value` non-comparable, because `Validator` can't implement `Ord` and `PartialOrd`.
`Token` contains list of parameters which are `Value`. So `Token` contains `Value` and can't implement `Ord` and `PartialOrd` too. So that I can't it in `BTreeSet`.

Reply:
> Can't we compare Token by it's definition_id?

I think everyone from the team asked this someday. No, because user can have two tokens with the same `definition_id`.
Example (pseudo json):

```
PermissionToken {
    definition_id: "can_set_key_value_in_user_asset",
    params: {
        "asset_id": "rose##alice@wonderland"
    },
},

PermissionToken {
    definition_id: "can_set_key_value_in_user_asset",
    params: {
        "asset_id": "tulip##bob@wonderland"
    },
},
```

Reply:
@Erigara please resolve this if you agree

Reply:
ah, got it